src/mme-app/contextManager/dataBlocks.cpp

```diff
@@ -628,20 +709,64 @@ namespace mme
 
 	
 	/******************************************************************************
-	* sets BearerContext
+	* add BearerContext
+	******************************************************************************/
+	bool SessionContext::addBearerContext(BearerContext* BearerContext_i)
+	{
+	        if( BearerContext_m.size() <= 15 )
+    		        BearerContext_m.push_back(BearerContext_i);
+    		else
+    		        return false;
+    		return true;
+	}
+	
+	/******************************************************************************
+	* remove BearerContext
+	******************************************************************************/
+	void SessionContext::removeBearerContext(BearerContext* BearerContext_i)
+	{
+    	    	BearerContext_m.remove(BearerContext_i);
+	}
+		
+	/******************************************************************************
+	* returns BearerContextList
+	******************************************************************************/
+	std::list<BearerContext*>& SessionContext::getBearerContextContainer()
+	{
+		return BearerContext_m;
+	}
+	
+	/******************************************************************************
+	* find BearerContext ByBearerId
 	******************************************************************************/
-	void SessionContext::setBearerContext( BearerContext* BearerContextp )
+	BearerContext* SessionContext::findBearerContextByBearerId( uint8_t bearerId_i)
```

Comment:
Should not this be array ? may be it would be faster.

Reply:
For array, we will have to pre-allocate 2 * 11 * 4 bytes for each UE for supporting 11 PDNS/11 bearers. I guess in most normal situations UE may have max 3 PDNs/bearers. That is why we looked at list/vector.